vignettes/demo.Rmd

```diff
@@ -229,66 +242,77 @@ fledge::finalize_version()
+{tea} with cup is tested, now we want to enhance with bowl.
 This requires changes to the code, and perhaps a new test.
-We create a branch and switch to this branch to implement this.
+We create a branch (whose name starts with a "f" for "feature") and switch to this branch to implement this.
 
-```{bash super-frob-branch}
-git checkout -b f-super-frob
+```{r bowl-branch}
+gert::git_branch_create("f-bowl", checkout = TRUE)
 ```
 
 On the branch, separate commits are used for the tests and the implementation.
 These commit messages do not need to be formatted specially, because {fledge} will ignore them anyway.
 
-```{r super-frob}
-usethis::use_test("frob")
-```
+This time we write the tests first, test-driven development.
 
+```{r bowl}
+usethis::use_test("bowl")
+```
 
-```{bash super-frob-commit}
-git add .
-git status
-git commit -m "Add super-frob tests."
+```{r bowl-git, results='hide'}
+gert::git_add("tests/testthat/test-bowl.R")
+gert::git_commit("Add bowl tests.")
 ```
 
 
-```{r super-frob-2}
-usethis::use_r("frob")
-writeLines("# super-frob", "R/frob.R")
+```{r bowl-2}
+usethis::use_r("bowl")
+writeLines("# bowl of tea", "R/bowl.R")
 ```
 
-
-```{bash super-frob-2-commit}
-git add .
-git status
-git commit -m "Add super-frob implementation."
+```{r bowl-2-git, results='hide'}
+gert::git_add("R/bowl.R")
+gert::git_commit("Add bowl implementation.")
 ```
 
 This branch can be pushed to the remote as usual.
 Only when merging we specify the message to be included in the changelog, again with a bullet.[^merge-ff]
+You might be used to doing the merges on a remote (e.g. GitHub pull requests) but here we demonstrate a local merge commit.
 
 [^merge-ff]: Note that we really need a merge commit here; the default is to fast-forward which doesn't give us the opportunity to insert the message intended for the changelog.
 Earlier, we set the `merge.ff` config option to `"false"` to achieve this.
 
-```{bash super-frob-merge}
-git checkout main
-git merge f-super-frob --no-commit
-git status
-git commit -m "- Super-frobnication enabled."
+```{r bowl-merge, results='hide'}
+gert::git_branch_checkout("main")
+gert::git_merge("f-bowl", commit = FALSE)
+gert::git_commit("- Bowl of tea enabled.")
 ```
 
 The same strategy can be used when merging a pull/merge/... request on GitHub, GitLab, ...: use bullet points in the merge commit message to indicate the items to include in `NEWS.md`.
 
-Now that we have added super-frobnication support to our package, it is time to bump the version again.
+Now that we have added bowl support to our package, it is time to bump the version again.
 
 ```{r bump-two}
 fledge::bump_version()
 news <- readLines("NEWS.md")
 cat(news, sep = "\n")
 ```
 
+It seems we do not even need to amend the `NEWS.md` by hand this time as we made no typo!
+
 ## Prepare for release
 
 After multiple cycles of development, review and testing, we decide that our package is ready for release to CRAN.
```

Comment:
Note to self: add an explicit mention to the fact is that the difference here is that you have to wait, some external system will determine whether the version is fit for release, and you might have to do some unexpected tweaks.